compiler/z/codegen/OpMemToMem.cpp

```diff
@@ -1514,32 +1528,19 @@ MemCmpVarLenMacroOp::generate(TR::Register* dstReg, TR::Register* srcReg, TR::Re
    _offset = offset;
    _cursor = cursor;
    _litReg = NULL;
-   TR::Compilation *comp = _cg->comp();
+   TR::LabelSymbol * cFlowRegionEnd = generateLabelSymbol(_cg);
 
-   if(cursorBefore == NULL) cursorBefore = _cg->getAppendInstruction();
+   generateShortCircuitInstructions(_cg, _rootNode, _resultReg, _srcReg, _dstReg, _doneLabel);
    generateLoop();
    setInRemainder(true);
    generateRemainder();
 
    _cursor = generateCmpResult(_cg, _rootNode, _dstReg, _srcReg, _resultReg, _falseLabel, _trueLabel, _doneLabel);
-   TR::RegisterDependencyConditions * dependencies = generateDependencies();
-   _cursor->setDependencyConditions(dependencies);
-   if(_startControlFlow==NULL)
-     {
-     _startControlFlow=cursorBefore->getNext();
-     if(_startControlFlow->getOpCodeValue() == TR::InstOpCode::assocreg) _startControlFlow=_startControlFlow->getNext();
-     }
-   if(_startControlFlow != _cursor)
-     {
-      TR::LabelSymbol * cFlowRegionStart = generateLabelSymbol(_cg);
-      TR::LabelSymbol * cFlowRegionEnd = generateLabelSymbol(_cg);
 
-      generateS390LabelInstruction(_cg, TR::InstOpCode::label, _rootNode, cFlowRegionStart, dependencies, _startControlFlow->getPrev());
-      cFlowRegionStart->setStartInternalControlFlow();
+   TR::RegisterDependencyConditions * dependencies = generateDependencies();
+   generateS390LabelInstruction(_cg, TR::InstOpCode::label, _rootNode, cFlowRegionEnd, dependencies);
```

Comment:
Control flow ends after generating results. In case of short circuiting, we skip generating results since result reg has the desired value of zero.